StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement+Internal.swift

```diff
@@ -228,9 +228,11 @@ extension EmbeddedPaymentElement: VerticalSavedPaymentMethodsViewControllerDeleg
             savedPaymentMethods: latestPaymentMethods
         )
 
-        // If there are still saved payment methods & the saved payment method was previously selected to presenting
+        // Select the selected payment method if we selected a payment method and have more than one saved PM
+        // Or if there are still saved payment methods & the saved payment method was previously selected to presenting
         // the list of saved payment methods, then the embedded view should continue to show it is selected, otherwise unselected.
-        let isSelected: Bool = latestPaymentMethods.count > 0 && embeddedPaymentMethodsView.selection?.isSaved ?? false
+        let isSelected = (latestPaymentMethods.count > 1 && selectedPaymentMethod != nil) ||
```

Comment:
is there any reason to check `latestPaymentMethods.count > 1` ? seems like we can just check `selectedPaymentMethod != nil || (embeddedPaymentMethodsView.selection?.isSaved ?? false && latestPaymentMethods.count > 0)`

Reply:
Yes we do need this! The saved payment screen when you delete all but the last payment method auto selects it, even if the user did not tap it. I think i this case we should not auto select that saved payment method in the embedded view. I extended the existing test to cover this now.

Reply:
Ah! right! thank you!